lib/src/ctx/context.rs

```diff
@@ -143,8 +143,17 @@ impl<'a> Context<'a> {
 	}
 
 	/// Add a timeout to the context. If the current timeout is sooner than
-	/// the provided timeout, this method does nothing.
+	/// the provided timeout, this method does nothing. If the result of the
+	/// addition causes an overflow, this method does nothing.
+	/// NOTE(gguillemas): This should probably return an option or a result
+	/// to be able to handle overflow cases transparently. This may be done
+	/// for v2.0.0, as it would change the public API.
```

Comment:
The `context` module is not documented, but is hidden from the docs and public API, so therefore we can update this now. Therefore it would probably be better to return and `Option` or `Result` for this function. What do you think?

Reply:
Perfect, I will look into how to best handle it considering that the signature can be changed 👍